### ISSUE TITLE
SSL icecast error cors

### ISSUE THREAD

**Issue by Reaktiv4** [AzuraCast/AzuraCast#6104]
### Installation Method

Docker Installation

### AzuraCast Release Channel

Rolling Release Channel

### Current AzuraCast Version

Rolling Release #919e3aa (2023-02-21 6:29)

### What happened?

New VPS installation. The certificate is installed and working, but the radio streams in icecast work on http.
Restarting and updating does not help.

As far as I understand the problem is that in the latest release of icecast the certificate and key must be in one file!!! And now they are in two files. The certificate is separate and the key is separate.
How can I fix the problem?

2) What is the full path in the system to the certificate and key? Where exactly are these files located on the system?

**Comment by Reaktiv4** [comment]
I reviewed all the files, I can’t find where the ssl files are located, I would manually make changes. Can anyone suggest?

**Comment by BusterNeece** [comment]
I can't reproduce this in my own instances. When I directly visit `radio-url:8000`, the SSL certificate works just fine there in its existing format, as a separate key/cert pair just like how nginx accepts it.

If you were looking to overwrite the certs we use, you would mount them using a `docker-compose.override.yml` file like this:

```yaml
services:
  web:
    volumes:
      - $PWD/your_ssl.crt/var/azuracast/acme/ssl.crt:ro
      - $PWD/your_ssl.key:/var/azuracast/acme/ssl.key:ro
```

**Comment by Reaktiv4** [comment]
It works fine directly, but if the stream is transferred to another domain, then there is a problem. The web browser accesses streams in icecast directly. If you go to the icecast administration page, you will see that the streams are in. I looked at the neighboring topics, all the problems of the cors are from here, as far as I can see. Azuracast uses a "mask https", but browsers access icecast directly. Go to your icecast administration page and see what format the links are in.

**Comment by BusterNeece** [comment]
> If you go to the icecast administration page, you will see that the streams are in

I don't understand what you mean by this.

I'm looking at the Icecast admin page on my local install and nothing seems wrong. There are no SSL certificate errors on this page. If there's a CORS problem, that's very likely upstream with Icecast, not with us.

**Comment by Reaktiv4** [comment]
Sorry, I didn't express myself clearly. Go to - https://demo.azuracast.com/listen/azuratest_radio/admin.html
Look at the page in what format the links are http or https ? How much have I watched everyone http.

**Comment by Reaktiv4** [comment]
Could it be a licensing issue? In the next topic, I added a link https://github.com/AzuraCast/AzuraCast/issues/6053#issuecomment-1445486954 
to an article by one of the experts, which I accidentally stumbled upon. Quote from there - Ubuntu/Debian do not supply a version of Icecast with HTTPS support enabled (due to some licensing issues), but Xiph (the foundation behind Icecast) supplies repositories with a pre-built version of Icecast with SSL support.
Could this be the problem?

**Comment by BusterNeece** [comment]
Our version of Icecast is custom-compiled from Icecast-KH. It absolutely has HTTPS support. The issue you're running into is likely caused by our web proxy settings, not anything more complex than that.

**Comment by Reaktiv4** [comment]
Tried in two different hosting companies, with different settings. There is this problem all the time. I look in neighboring topics and see that they also have - listenurl  http
One example is - https://github.com/AzuraCast/AzuraCast/issues/6019#issuecomment-1409025199
I've been setting it up for a few days now. I don't know what to do next.

**Comment by BusterNeece** [comment]
Wait, now you're saying the _listenurl_ is being set incorrectly?

You haven't been very clear at all about what specifically is the bug you're experiencing.

There's no need to try installing it on different providers or anything...it's likely you're encountering a legitimate bug, but it's very hard for me to follow what is actually going on enough to track it down and fix it.

**Comment by Reaktiv4** [comment]
I'm sorry for bad English. I'll try again. In the azuracast environment, that is, on the same domain, everything works fine. But, as soon as we add the stream on another domain, that is, we try to listen to the radio from another domain, it shows the error cors. In all browsers. Because the browser requests data from Azuracast and receives it via http. Although Azuracast has a certificate and links in https, but in icecast they are in http. And any browser is looking for the original link and stumbles upon http in icecast. Therefore the cors is an error. icecast links (streams) do not work on https, that's the whole problem. An example that they are in http I showed above. The problem is in the transfer of data to another domain. The radio works, the sound goes on, but there are cors errors in the browser due to the problem that I wrote about.
